From 876417ad29488376ab843c2440b3c90f53f25f2c Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 03 Mar 2024 11:52:34 +0200
Subject: [PATCH] [SW-0] package: agent-hbm: adjust make

The automation of Version creation was missed in the transition

Change-Id: Ib004183ff089b836adddbff219194c2712424026
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/agent-hbm/agent-hbm.mk b/package/agent-hbm/agent-hbm.mk
index 22fb7b3..4918c3a 100644
--- a/package/agent-hbm/agent-hbm.mk
+++ b/package/agent-hbm/agent-hbm.mk
@@ -99,8 +99,19 @@
 	@$(foreach module,$(AGENT_HBM_APP_MODULES),$(call AGENT_HBM_MODULE_VERSION_HOOK,$(module)))
 endef
 
+define AGENT_HBM_CREATE_HL_VERSION
+	@mkdir -p $(AGENT_HBM_BUILD_DIR)/include
+	@echo "#ifndef _HL_VERSION_" >  $(AGENT_HBM_BUILD_DIR)/include/_hl_version.h
+	@echo "#define _HL_VERSION_" >> $(AGENT_HBM_BUILD_DIR)/include/_hl_version.h
+	@echo "#define HL_VERSION_MAJOR $$(echo $(AGENT_HBM_LOCAL_VERSION) | cut -d. -f1)" >> $(AGENT_HBM_BUILD_DIR)/include/_hl_version.h
+	@echo "#define HL_VERSION_MINOR $$(echo $(AGENT_HBM_LOCAL_VERSION) | cut -d. -f2)" >> $(AGENT_HBM_BUILD_DIR)/include/_hl_version.h
+	@echo "#define HL_VERSION_PATCH $$(echo $(AGENT_HBM_LOCAL_VERSION) | cut -d. -f3)" >> $(AGENT_HBM_BUILD_DIR)/include/_hl_version.h
+	@echo "#endif" >> $(AGENT_HBM_BUILD_DIR)/include/_hl_version.h
+endef
+
 define AGENT_HBM_CONFIGURE_CMDS
 	$(AGENT_HBM_CONF_ENV) $(HOST_DIR)/bin/cmake $(AGENT_HBM_CONF_OPTS) -S $(AGENT_HBM_BUILD_DIR)/margin-tools/hbm -B $(AGENT_HBM_BUILD_DIR)/build
+	@$(call AGENT_HBM_CREATE_HL_VERSION)
 endef
 
 AGENT_HBM_POST_RSYNC_HOOKS += HL_VERSION_GEN
